src/infrastructure/discord/DiscordService.ts

```diff
@@ -60,4 +119,10 @@ export class DiscordService
   public onUserLeave(handler: (member: GuildMember) => void): void {
     this.onUserLeaveHandlers.push(handler);
   }
+
+  public onVoiceEvent(
+    handler: (event: DiscordGuildScheduledEvent) => void,
```

Comment:
seguindo o padrão dos outros handlers acredito que o DiscordService deveria manipular os tipos da sdk do discord e não precisa saber do formato da entity

além disso olhando o que temos hoje o registro do handler e o mapeamento é feito em um "command" que recebe os casos de uso, por exemplo UserCommand https://github.com/Coletivo-Popular-Design-Desenvolvimento/checkin-discord-bot-v1/blob/639ccc32e87ff2211949bae77a9e771907f53318/src/application/command/userCommand.ts#L85

então no meu entendimento
- `VoiceEventService` deveria ser `VoiceEventCommand`
- o registro do evento deveria ocorrer dentro desse command no lugar do app.context.ts
- o `convertDiscordEvent` deveria ser movido para dentro desse command e usar o tipo de entity
- depois disso acho que não vamos precisar mais do DiscordEventTypes.ts

```suggestion
    handler: (event: GuildScheduledEvent | PartialGuildScheduledEvent) => void,
```